src/main/java/org/hackedtogether/xpadditions/data/ModLootTableProvider.java

```diff
@@ -0,0 +1,52 @@
+package org.hackedtogether.xpadditions.data;
+
+import com.google.common.collect.ImmutableList;
+import com.mojang.datafixers.util.Pair;
+import net.minecraft.block.Block;
+import net.minecraft.data.DataGenerator;
+import net.minecraft.data.LootTableProvider;
+import net.minecraft.data.loot.BlockLootTables;
+import net.minecraft.loot.*;
+import net.minecraft.util.ResourceLocation;
+import net.minecraftforge.fml.RegistryObject;
+import org.hackedtogether.xpadditions.setup.ModBlocks;
+import org.hackedtogether.xpadditions.setup.Registration;
+
+import java.util.List;
+import java.util.Map;
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+public class ModLootTableProvider extends LootTableProvider {
+    public ModLootTableProvider(DataGenerator generator) {
+        super(generator);
+    }
+
+    @Override
+    protected List<Pair<Supplier<Consumer<BiConsumer<ResourceLocation, LootTable.Builder>>>, LootParameterSet>> getTables() {
+        return ImmutableList.of(
+                Pair.of(ModBlockLootTables::new, LootParameterSets.BLOCK)
+        );
+    }
+
+    @Override
+    protected void validate(Map<ResourceLocation, LootTable> map, ValidationTracker validationtracker) {
+        map.forEach((p_218436_2_, p_218436_3_) -> LootTableManager.validate(validationtracker, p_218436_2_, p_218436_3_));
```

Comment:
Shall we change the 2 `p_` vars names to `resourceLocation` and `lootTable` since they are just the elements in the `Map`?